File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = [0]
    def back_track(ct, s):
        if(ct==len(numbers)):
            if(s== target):
                answer[0]+=1
            return
        back_track(ct+1,s+numbers[ct])
        back_track(ct+1,s-numbers[ct])
    back_track(0,0)
    return answer[0]

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    public void btk(int[] numbers, int idx, int cur, int target){
        if(idx == numbers.length){
            if(cur == target){
                answer++;
            }
            return;
        }
        btk(numbers, idx+1, cur+ numbers[idx], target);
        btk(numbers, idx+1, cur- numbers[idx], target);
    }
    public int solution(int[] numbers, int target) {
        btk(numbers, 0, 0, target);
        return answer;
    }
}